### PR TITLE
fix(Separator): fix background and border issue

### PR DESCRIPTION
**Pull request by mainframev**
[Reported in Slack](https://skypicker.slack.com/archives/CAMS40F7B/p1689588311958009)

### CODE REVIEW

File: packages/orbit-components/src/Separator/types.d.ts [hunk 1]
```diff
@@ -83,9 +83,19 @@ export type PaletteTokens =
...
+export interface CommonProps extends Common.SpaceAfter {
   indent?: Indent;
-  type?: "solid" | "dashed" | "dotted" | "double" | "none";
-  color?: PaletteTokens;
   align?: Align;
 }
+
+type ConditionalSeparatorType =
+  | {
+      readonly color: PaletteTokens;
```
Comment (DSil) [inline packages/orbit-components/src/Separator/types.d.ts:93]: I think we want this
```suggestion
      readonly color?: PaletteTokens;
```

In the past, we were able to have a type without defining a color. If we don't make the `color` optional here, there will be an error requiring `color` to be defined.

**Review by DSil — CHANGES_REQUESTED** [review]

File: packages/orbit-components/src/Separator/TYPESCRIPT_TEMPLATE.template [hunk 2]
```diff
@@ -7,9 +7,19 @@ export type Align = "left" | "center" | "right";
...
+export interface CommonProps extends Common.SpaceAfter {
   indent?: Indent;
-  type?: "solid" | "dashed" | "dotted" | "double" | "none";
-  color?: PaletteTokens;
   align?: Align;
 }
+
+type ConditionalSeparatorType =
+  | {
+      readonly color: PaletteTokens;
```
Comment (DSil) [inline packages/orbit-components/src/Separator/TYPESCRIPT_TEMPLATE.template:17]: I didn't know you'd accept the suggestion directly here 😅 so this one needs a change as well
```suggestion
      readonly color?: PaletteTokens;
```

Comment (mainframev) [reply packages/orbit-components/src/Separator/TYPESCRIPT_TEMPLATE.template:17]: RIght :) 👍🏻

**Review by DSil — APPROVED** [review]
Squash commits and 🚀 👍

**Pull request merged**